lib/src/model/container.dart

```diff
@@ -5,121 +5,215 @@
+
+  @nonVirtual
+  bool get hasPublicInstanceOperators =>
+      publicInstanceOperatorsSorted.isNotEmpty;
+
+  @nonVirtual
+  Iterable<Operator> get publicInstanceOperators =>
+      model_utils.filterNonPublic(instanceOperators);
+
+  List<Operator> _publicInstanceOperatorsSorted;
+  List<Operator> get publicInstanceOperatorsSorted =>
+      _publicInstanceOperatorsSorted ??= publicInstanceOperators.toList()
+        ..sort(byName);
+
+  /// Fields fully declared in this [Container].
+  Iterable<Field> get declaredFields;
+
+  /// All fields accessible in this instance that are not static.
+  Iterable<Field> get instanceFields =>
+      declaredFields.where((f) => !f.isStatic);
+
+  bool get hasInstanceFields => instanceFields.isNotEmpty;
+
+  @nonVirtual
+  Iterable<Field> get publicInstanceFields =>
+      model_utils.filterNonPublic(instanceFields);
+
+  @nonVirtual
+  bool get hasPublicInstanceFields => publicInstanceFields.isNotEmpty;
+
+  List<Field> _publicInstanceFieldsSorted;
+  List<Field> get publicInstanceFieldsSorted => _publicInstanceFieldsSorted ??=
+      publicInstanceFields.toList()..sort(byName);
+
+  Iterable<Field> get constantFields => declaredFields.where((f) => f.isConst);
+
+  Iterable<Field> get publicConstantFields =>
+      model_utils.filterNonPublic(constantFields);
+
+  bool get hasPublicConstantFields => publicConstantFieldsSorted.isNotEmpty;
+
+  List<Field> _publicConstantFieldsSorted;
+  List<Field> get publicConstantFieldsSorted => _publicConstantFieldsSorted ??=
+      publicConstantFields.toList()..sort(byName);
+
+  Iterable<Accessor> get instanceAccessors =>
+      instanceFields.expand((f) => f.allAccessors);
+
+  Iterable<Accessor> get staticAccessors =>
+      staticFields.expand((f) => f.allAccessors);
+
+  /// This container might be canonical for elements it does not contain.
+  /// See [Inheritable.canonicalEnclosingContainer].
+  bool containsElement(Element element) => allElements.contains(element);
+
+  Set<Element> _allElements;
+  Set<Element> get allElements =>
+      _allElements ??= allModelElements.map((e) => e.element).toSet();
+
+  Map<String, List<ModelElement>> _membersByName;
+
+  /// Given a ModelElement that is a member of some other class, return
+  /// a member of this class that has the same name and return type.
+  ///
+  /// This enables object substitution for canonicalization, such as Interceptor
+  /// for Object.
+  ModelElement memberByExample(ModelElement example) {
+    if (_membersByName == null) {
+      _membersByName = {};
+      for (var me in allModelElements) {
+        if (!_membersByName.containsKey(me.name)) {
+          _membersByName[me.name] = [];
+        }
+        _membersByName[me.name].add(me);
+      }
+    }
+    ModelElement member;
+    var possibleMembers = _membersByName[example.name]
+        .where((e) => e.runtimeType == example.runtimeType);
+    if (example.runtimeType == Accessor) {
+      possibleMembers = possibleMembers.where(
```

Comment:
Can this code be updated to use whereType? It seems very strange to use runtimeType here...